AirportTicketBooking/Utility/GetUserInput.cs

```diff
@@ -0,0 +1,227 @@
+using System;
+using System.Text.RegularExpressions;
+using AirportTicketBooking.Enums;
+
+namespace AirportTicketBooking.Utility
+{
+    public static class GetUserInput
+    {
+        private static readonly Random Random = new Random();
+        
+        public static decimal GetDecimalData(string name)
+        {
+            Console.WriteLine($"Please specify the {name} you want to filter upon. Price must be of this format [Whole Numbers].[Decimal Number]");
+            var choiceTest = Console.ReadLine();
+            decimal num;
+            while (!decimal.TryParse(choiceTest, out num))
+            {
+                Console.WriteLine("Please specify a valid decimal number!");
+            }
+
+            return num;
+        }
+
+        public static string GetStringData(string name)
+        {
+            Console.WriteLine($"Please specify the {name}.");
+            var value =  Console.ReadLine();
+            while (string.IsNullOrWhiteSpace(value))
+            {
+                Console.WriteLine("Please specify a valid string value!");
+                value = Console.ReadLine();
+            }
+
+            return value;
+        }
+
+        public static string GetDateOnlyData(string name)
+        {
+            Console.WriteLine($"Please specify the {name} you want to filter upon.");
+            Console.WriteLine("Date values must be of this format: mm/dd/yyyy.");
+            var value =  Console.ReadLine();
+            while (value?.Split('/').Length < 0 || string.IsNullOrWhiteSpace(value))
+            {
+                Console.WriteLine("Please specify a valid date value!");
+                value = Console.ReadLine();
+            }
+
+            return value;
+        }
+
+        public static int GetIdFromUser()
+        {
+            Console.WriteLine("Please specify your preferred id, hit enter if you can't decide.");
+            var strId = Console.ReadLine();
+            var id = string.IsNullOrWhiteSpace(strId) ? Random.Next(100, 1000) : int.Parse(strId);
+
+            return id;
+        }
+
+        public static int GetIntegerFromUser(string name)
+        {
+            Console.WriteLine($"Please specify the {name}");
+            var choiceTest = Console.ReadLine();
+            int num;
+            while (!int.TryParse(choiceTest, out num))
+            {
+                Console.WriteLine("Please specify a valid integer number!");
+            }
```

Comment:
you can use           `  while (!int.TryParse(choiceTest, out int num))`

instead of declaring the variable separately

Reply:
The variable won't be visible to the lines outside of the `int.TryParse()` duo to scooping.